cli/compose/types/types.go

```diff
@@ -70,14 +141,67 @@ func (cd ConfigDetails) LookupEnv(key string) (string, bool) {
 
 // Config is a full compose file configuration
 type Config struct {
-	Filename string `yaml:"-"`
-	Version  string
-	Services Services
-	Networks map[string]NetworkConfig   `yaml:",omitempty"`
-	Volumes  map[string]VolumeConfig    `yaml:",omitempty"`
-	Secrets  map[string]SecretConfig    `yaml:",omitempty"`
-	Configs  map[string]ConfigObjConfig `yaml:",omitempty"`
-	Extras   map[string]interface{}     `yaml:",inline"`
+	Filename string                     `yaml:"-" json:"-"`
+	Version  string                     `json:"version"`
+	Services Services                   `json:"services"`
+	Networks map[string]NetworkConfig   `yaml:",omitempty" json:"networks,omitempty"`
+	Volumes  map[string]VolumeConfig    `yaml:",omitempty" json:"volumes,omitempty"`
+	Secrets  map[string]SecretConfig    `yaml:",omitempty" json:"secrets,omitempty"`
+	Configs  map[string]ConfigObjConfig `yaml:",omitempty" json:"configs,omitempty"`
+	Extras   map[string]interface{}     `yaml:",inline", json:"-"`
+}
+
+// MarshalJSON makes Config implement json.Marshaler
+func (c Config) MarshalJSON() ([]byte, error) {
```

Comment:
I don't understand why we need this function... I think the struct field tags you added should produce an equivalent output with the default marshaller.

Reply:
This is necessary if we want to include top-level extension fields in the output, as there is no equivalent to the `yaml:inline` anotation for JSON. If we're okay with those being omitted from the JSON output, it can be removed.